internal/controllers/catalog_test.go

```diff
@@ -0,0 +1,94 @@
+package controllers_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	"github.com/operator-framework/operator-controller/internal/controllers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func collectRequests(reqChan <-chan string) func() []string {
+	var reqNames []string
+	return func() []string {
+		select {
+		case req := <-reqChan:
+			reqNames = append(reqNames, req)
+		default:
+		}
+		return reqNames
+	}
+}
+
+var _ = Describe("SetupWithManager", func() {
+	When("catalog events occur", func() {
+		var cancel context.CancelFunc
+		var ctx context.Context
+		var opNames []string
+		var reqChan chan string
+		BeforeEach(func() {
+			ctx = context.Background()
+			mgr, err := ctrl.NewManager(cfg, ctrl.Options{Scheme: sch})
+			Expect(err).To(BeNil())
+
+			opNames = []string{"prometheus", "project-quay"}
+			reqChan = make(chan string, 4*len(opNames))
```

Comment:
Do we still need the buffer to be `4*len(opNames)`? I don't think its a huge deal, but IIUC we could probably reduce this to just `len(opNames)` since a receiver will be using the channel after each "reconcile" that occurs.